components/eamxx/src/share/field/field_utils_impl.hpp

```diff
@@ -214,6 +214,72 @@ void randomize (const Field& f, Engine& engine, PDF&& pdf)
   f.sync_to_dev();
 }
 
+template<typename ST, typename Engine, typename PDF, typename MaskType>
+void perturb (const Field& f,
+              Engine& engine,
+              PDF&& pdf,
+              const unsigned int base_seed,
+              const MaskType& level_mask,
+              const Field& dof_gids)
+{
+  const auto& fl = f.get_header().get_identifier().get_layout();
+
+  // Check to see if field has a column dimension
+  using namespace ShortFieldTagsNames;
+  const bool has_column_dim = fl.has_tag(COL);
+
+  if (has_column_dim) {
+    // Because Column is the partitioned dimension, we must reset the
+    // RNG seed to be the same on every column so that a column will
+    // have the same value no matter where it exists in an MPI rank's
+    // set of local columns.
+    dof_gids.sync_to_host();
```

Comment:
No need to sync gids to host. They are immutable during a simulation, so we make sure they are synced when we construct the grid.